File: phenomenal/strategy_50.rb
class BaseClass
  def test
    raise "Abstract base class method"
  end
end

class Strategy1 < BaseClass
  def test
    1
  end
end 
class Strategy2 < BaseClass
  def test
    2
  end
end 
class Strategy3 < BaseClass
  def test
    3
  end
end 
class Strategy4 < BaseClass
  def test
    4
  end
end 
class Strategy5 < BaseClass
  def test
    5
  end
end 
class Strategy6 < BaseClass
  def test
    6
  end
end 
class Strategy7 < BaseClass
  def test
    7
  end
end 
class Strategy8 < BaseClass
  def test
    8
  end
end 
class Strategy9 < BaseClass
  def test
    9
  end
end 
class Strategy10 < BaseClass
  def test
    10
  end
end 
class Strategy11 < BaseClass
  def test
    11
  end
end 
class Strategy12 < BaseClass
  def test
    12
  end
end 
class Strategy13 < BaseClass
  def test
    13
  end
end 
class Strategy14 < BaseClass
  def test
    14
  end
end 
class Strategy15 < BaseClass
  def test
    15
  end
end 
class Strategy16 < BaseClass
  def test
    16
  end
end 
class Strategy17 < BaseClass
  def test
    17
  end
end 
class Strategy18 < BaseClass
  def test
    18
  end
end 
class Strategy19 < BaseClass
  def test
    19
  end
end 
class Strategy20 < BaseClass
  def test
    20
  end
end 
class Strategy21 < BaseClass
  def test
    21
  end
end 
class Strategy22 < BaseClass
  def test
    22
  end
end 
class Strategy23 < BaseClass
  def test
    23
  end
end 
class Strategy24 < BaseClass
  def test
    24
  end
end 
class Strategy25 < BaseClass
  def test
    25
  end
end 
class Strategy26 < BaseClass
  def test
    26
  end
end 
class Strategy27 < BaseClass
  def test
    27
  end
end 
class Strategy28 < BaseClass
  def test
    28
  end
end 
class Strategy29 < BaseClass
  def test
    29
  end
end 
class Strategy30 < BaseClass
  def test
    30
  end
end 
class Strategy31 < BaseClass
  def test
    31
  end
end 
class Strategy32 < BaseClass
  def test
    32
  end
end 
class Strategy33 < BaseClass
  def test
    33
  end
end 
class Strategy34 < BaseClass
  def test
    34
  end
end 
class Strategy35 < BaseClass
  def test
    35
  end
end 
class Strategy36 < BaseClass
  def test
    36
  end
end 
class Strategy37 < BaseClass
  def test
    37
  end
end 
class Strategy38 < BaseClass
  def test
    38
  end
end 
class Strategy39 < BaseClass
  def test
    39
  end
end 
class Strategy40 < BaseClass
  def test
    40
  end
end 
class Strategy41 < BaseClass
  def test
    41
  end
end 
class Strategy42 < BaseClass
  def test
    42
  end
end 
class Strategy43 < BaseClass
  def test
    43
  end
end 
class Strategy44 < BaseClass
  def test
    44
  end
end 
class Strategy45 < BaseClass
  def test
    45
  end
end 
class Strategy46 < BaseClass
  def test
    46
  end
end 
class Strategy47 < BaseClass
  def test
    47
  end
end 
class Strategy48 < BaseClass
  def test
    48
  end
end 
class Strategy49 < BaseClass
  def test
    49
  end
end 
class Strategy50 < BaseClass
  def test
    50
  end
end 
class Foo < BaseClass
  def initialize strategy
    @strategy = strategy
  end
  def set_strategy new_strategy
    @strategy = new_strategy
  end
  def test
    @strategy.test
  end
end

def testContexts
  $i = 0
  $ITER = 1000
  $CALLS = 100
  beginning_time = Time.now
  foo = Foo.new(Strategy1.new)
  until $i > $ITER do
    num = 1 + rand(50)
    j = 0
    foo.set_strategy(Object.const_get("Strategy#{num}").new)
    for j in (1..$CALLS)
      foo.test
    end
    $i +=1;
  end
  return (Time.now - beginning_time)*1000
end

def runs(run)
  $times = 0
  File.open("runs_strategy/50/test#{run}.csv", 'w') {|f| f.write("iteration, Elapsed time in milliseconds
")}
  for $times in 1..100
    elapsed_time = testContexts
    File.open("runs_strategy/50/test#{run}.csv", 'a') do |f|
        f.puts("#{$times}, #{elapsed_time}")
      end
  end
end

for run in 1..10
  runs(run)
end

File: Subjective-C/strategy/strategy_50.m
#import <Foundation/Foundation.h>
#include <stdlib.h>
#import "strategy_50.h" 

@implementation BaseClass
-(id) init{
    return self = [super init];
}; 
-(int) testS {
    return 0;
}
@end 

@implementation Strategy1
-(int) testS {
    return 1;
}
@end 

@implementation Strategy2
-(int) testS {
    return 2;
}
@end 

@implementation Strategy3
-(int) testS {
    return 3;
}
@end 

@implementation Strategy4
-(int) testS {
    return 4;
}
@end 

@implementation Strategy5
-(int) testS {
    return 5;
}
@end 

@implementation Strategy6
-(int) testS {
    return 6;
}
@end 

@implementation Strategy7
-(int) testS {
    return 7;
}
@end 

@implementation Strategy8
-(int) testS {
    return 8;
}
@end 

@implementation Strategy9
-(int) testS {
    return 9;
}
@end 

@implementation Strategy10
-(int) testS {
    return 10;
}
@end 

@implementation Strategy11
-(int) testS {
    return 11;
}
@end 

@implementation Strategy12
-(int) testS {
    return 12;
}
@end 

@implementation Strategy13
-(int) testS {
    return 13;
}
@end 

@implementation Strategy14
-(int) testS {
    return 14;
}
@end 

@implementation Strategy15
-(int) testS {
    return 15;
}
@end 

@implementation Strategy16
-(int) testS {
    return 16;
}
@end 

@implementation Strategy17
-(int) testS {
    return 17;
}
@end 

@implementation Strategy18
-(int) testS {
    return 18;
}
@end 

@implementation Strategy19
-(int) testS {
    return 19;
}
@end 

@implementation Strategy20
-(int) testS {
    return 20;
}
@end 

@implementation Strategy21
-(int) testS {
    return 21;
}
@end 

@implementation Strategy22
-(int) testS {
    return 22;
}
@end 

@implementation Strategy23
-(int) testS {
    return 23;
}
@end 

@implementation Strategy24
-(int) testS {
    return 24;
}
@end 

@implementation Strategy25
-(int) testS {
    return 25;
}
@end 

@implementation Strategy26
-(int) testS {
    return 26;
}
@end 

@implementation Strategy27
-(int) testS {
    return 27;
}
@end 

@implementation Strategy28
-(int) testS {
    return 28;
}
@end 

@implementation Strategy29
-(int) testS {
    return 29;
}
@end 

@implementation Strategy30
-(int) testS {
    return 30;
}
@end 

@implementation Strategy31
-(int) testS {
    return 31;
}
@end 

@implementation Strategy32
-(int) testS {
    return 32;
}
@end 

@implementation Strategy33
-(int) testS {
    return 33;
}
@end 

@implementation Strategy34
-(int) testS {
    return 34;
}
@end 

@implementation Strategy35
-(int) testS {
    return 35;
}
@end 

@implementation Strategy36
-(int) testS {
    return 36;
}
@end 

@implementation Strategy37
-(int) testS {
    return 37;
}
@end 

@implementation Strategy38
-(int) testS {
    return 38;
}
@end 

@implementation Strategy39
-(int) testS {
    return 39;
}
@end 

@implementation Strategy40
-(int) testS {
    return 40;
}
@end 

@implementation Strategy41
-(int) testS {
    return 41;
}
@end 

@implementation Strategy42
-(int) testS {
    return 42;
}
@end 

@implementation Strategy43
-(int) testS {
    return 43;
}
@end 

@implementation Strategy44
-(int) testS {
    return 44;
}
@end 

@implementation Strategy45
-(int) testS {
    return 45;
}
@end 

@implementation Strategy46
-(int) testS {
    return 46;
}
@end 

@implementation Strategy47
-(int) testS {
    return 47;
}
@end 

@implementation Strategy48
-(int) testS {
    return 48;
}
@end 

@implementation Strategy49
-(int) testS {
    return 49;
}
@end 

@implementation Strategy50
-(int) testS {
    return 50;
}
@end 

const static int ITER = 1000;
const static int CALLS = 100; 

@implementation Foo 
@synthesize strategy;
-(int) testS {
    return [self.strategy testS];
} 

-(double) testContexts {
    NSDate *start = [NSDate date];
    for(int i = 0; i < ITER; i++) {
        int num = 1 + arc4random_uniform(50);
        [self setStrategy: [[NSClassFromString([NSString stringWithFormat:@"Strategy%d", num])alloc] init]];
        for(int j = 1; j <= CALLS; j++) {
            [self testS];
        }
    }
    NSTimeInterval timeInterval = fabs([start timeIntervalSinceNow]);
    return timeInterval*1000;
} 

-(void) runs:(int) run {
    NSMutableString *data = [NSMutableString stringWithString:@"Iterations, Elapsed time in milliseconds \n "];
    for(int times = 1; times <= 100; times++) {
        double elapsed_time = [self testContexts];
        [data appendFormat:@"%d, %f \n", times, elapsed_time];
    }
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *documentsDirectory = [paths objectAtIndex:0];
    NSString *fileName = [NSString stringWithFormat:@"%@/runs_strategy/50/test%d.csv",documentsDirectory, run];
    NSFileManager *fileManager = [NSFileManager defaultManager];
    if (![fileManager fileExistsAtPath:fileName]) {
        BOOL success = [[NSString stringWithString: data] writeToFile:fileName atomically:YES encoding:NSUTF8StringEncoding error:nil];
        if (!success) {
            NSLog(@"Error writting file");
        }
    } else {
        NSFileHandle *fileHandle = [NSFileHandle fileHandleForWritingAtPath:fileName];
        NSData *textData = [data dataUsingEncoding:NSUTF8StringEncoding];
        [fileHandle writeData:textData];
        [fileHandle closeFile];
    }
}
@end

int main(int argc, const char * argv[]) {
    Foo *foo = [[Foo alloc] init];
    for(int run=1; run <=10; run++) {
        [foo runs:run];
     }
    return 0;
}
